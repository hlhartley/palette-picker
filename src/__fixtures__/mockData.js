export const mockProjects = [
    {
        id: 1,
        name: "Warm colors project",
        created_at: "2019-03-26T05:17:02.689Z",
        updated_at: "2019-03-26T05:17:02.689Z"
    },
    {
        id: 2,
        name: "Cool colors project",
        created_at: "2019-03-26T05:17:02.702Z",
        updated_at: "2019-03-26T05:17:02.702Z"
    },
    {
        id: 3,
        name: "Neutral colors project",
        created_at: "2019-03-26T05:17:02.727Z",
        updated_at: "2019-03-26T05:17:02.727Z"
    }
]

export const mockPalettes = [
    {
        id: 1,
        project_id: 1,
        name: "Summer colors",
        color1: "FFFF82",
        color2: "DDDCF7",
        color3: "B5D99C",
        color4: "FFAF66",
        color5: "E65F5C",
        created_at: "2019-03-26T05:17:02.735Z",
        updated_at: "2019-03-26T05:17:02.735Z"
    },
    {
        id: 2,
        project_id: 1,
        name: "Spring colors",
        color1: "F25469",
        color2: "BD2D87",
        color3: "D664BE",
        color4: "DF99F0",
        color5: "B191FF",
        created_at: "2019-03-26T05:17:02.736Z",
        updated_at: "2019-03-26T05:17:02.736Z"
    },
    {
        id: 3,
        project_id: 2,
        name: "Fall colors",
        color1: "FCD0A1",
        color2: "B1B695",
        color3: "A690A4",
        color4: "EFA958",
        color5: "AFD2E9",
        created_at: "2019-03-26T05:17:02.751Z",
        updated_at: "2019-03-26T05:17:02.751Z"
    },
    {
        id: 4,
        project_id: 3,
        name: "Winter colors",
        color1: "594157",
        color2: "726DA8",
        color3: "7D8CC4",
        color4: "A0D2DB",
        color5: "BEE7E8",
        created_at: "2019-03-26T05:17:02.757Z",
        updated_at: "2019-03-26T05:17:02.757Z"
    }
]

export const mockHexcodes = [
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    },
    {
        color: 'ffffff',
        isLocked: false
    }
]