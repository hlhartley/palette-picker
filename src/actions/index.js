export const setError = (error) => ({
  type: 'SET_ERROR', 
  error
});

export const setLoading = (isLoading) => ({
  type: 'SET_LOADING', 
  isLoading
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS', 
  projects
});

export const setPalettes = (palettes) => ({
  type: 'SET_PALETTES', 
  palettes
});

export const setCurrentPaletteId = (paletteId) => ({
  type: 'SET_CURRENT_PALETTE_ID', 
  paletteId
});

export const setCurrentProjectName = (projectName) => ({
  type: 'SET_CURRENT_PROJECT_NAME', 
  projectName
});

export const setRandomHexcodes = (hexcodes) => ({
  type: 'SET_RANDOM_HEXCODES', 
  hexcodes
});

export const toggleDisplayRandom = (shouldDisplay) => ({
  type: 'DISPLAY_RANDOM', 
  shouldDisplay
});
