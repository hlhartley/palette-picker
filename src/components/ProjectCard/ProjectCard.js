import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEdit } from '@fortawesome/free-solid-svg-icons';
import { faTrash } from '@fortawesome/free-solid-svg-icons';

export class ProjectCard extends Component {
  render() {
    return (
      <div className="ProjectCard">
        <p className="project-name">Project Name</p>
          <div className="palette-container">
            <p className="palette-name">Palette Name</p>
              <div className="palette-minibox">colors</div>
              <button className="load-button"><FontAwesomeIcon icon={faEdit} className="load-icon"/></button>
              <button className="delete-button"><FontAwesomeIcon icon={faTrash} className="deelete-icon"/></button>
          </div>
      </div>
    );
  }
}

export default ProjectCard;
