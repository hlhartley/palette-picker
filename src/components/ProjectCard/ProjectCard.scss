.ProjectCard {
    border: 1px solid purple;
    padding: 20px;
}

.project-name {
    font-size: 18px;
    font-weight: bold;
    text-decoration: underline;
    padding-bottom: 10px;
}

.palette-name {
    font-size: 18px;
    margin-right: 10px;
}

.mapped-palettes {
    margin-bottom: 10px;
    display: flex;
}

.delete-project-button {
    margin-top: 10px;
    height: 30px;
    min-width: 88px;
    padding: 3px 8px;
    line-height: 1.5;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    touch-action: manipulation;
    cursor: pointer;
    border:0;
    border-radius: 2px;
    background: #9c27b0;
    color:#fff;
    outline:0;
    &:hover {
        background: darken(#9c27b0, 12%)  
    }
}

.md-btn-raised {
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 
                0 1px 5px 0 rgba(0, 0, 0, 0.12), 
                0 3px 1px -2px rgba(0, 0, 0, 0.2);
    transition: box-shadow 0.28s cubic-bezier(0.4, 0, 0.2, 1);

    &:active {
        box-shadow: 0 8px 10px 1px rgba(0, 0, 0, 0.14),
                0 3px 14px 2px rgba(0, 0, 0, 0.12),
                0 5px 5px -3px rgba(0, 0, 0, 0.4);
    }
}

.color-minibox {
    padding-left: 14px;
    padding-right: 14px;
}

.load-button,
.delete-button {
    font-size: 18px;
    border: none;
    margin-left: 3px;
    color: grey;
}