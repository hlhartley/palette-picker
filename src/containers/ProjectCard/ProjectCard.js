import React, { Component } from 'react';
import { connect } from 'react-redux';
import PaletteCard from '../../containers/PaletteCard/PaletteCard';
import { fetchData } from '../../utils/api.js';
import { setLoadedProject, deleteProject } from '../../actions';

export class ProjectCard extends Component {

  findProjectPalettes = () => {
    const  { palettes, project } = this.props;
      const matchingPalettes = palettes.filter((palette) => {
        return palette.project_id === project.id
    })
    return matchingPalettes;
  }


  deleteProject = async () => {
    let projectId = this.props.project.id;
    let response = await fetchData(`/projects/${projectId}`, 'DELETE');
    let IdToDelete = response.id;
    this.props.deleteProject(IdToDelete)
  }

  render() {
    const { project } = this.props;
    return (
      <div className="ProjectCard">
          <div onClick={() => this.props.setLoadedProject(this.props.project.name)}>
            <p className="project-name">{project.name.toUpperCase()}</p>
              <div className="palette-container">
                <PaletteCard projectPalettes={this.findProjectPalettes()}/>
              </div>
            <button onClick={this.deleteProject} className="delete-project-button">Delete Project</button>
          </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  palettes: state.palettes
});

export const mapDispatchToProps = (dispatch) => ({
  setLoadedProject: (loadedProject) => dispatch(setLoadedProject(loadedProject)),
  deleteProject: (id) => dispatch(deleteProject(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(ProjectCard);