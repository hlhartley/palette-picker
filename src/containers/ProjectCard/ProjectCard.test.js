import React from 'react';
import { shallow } from 'enzyme';
import { ProjectCard, mapStateToProps, mapDispatchToProps } from './ProjectCard';
import { setLoadedProject } from '../../actions';
import { mockPalettes } from '../../__fixtures__/mockData';

describe('ProjectCard', () => {
  let wrapper;
  let mockSetLoadedProject;
  beforeEach(() => {
    wrapper = shallow(
      <ProjectCard setLoadedProject={mockSetLoadedProject} project={{id: 1, name: 'project1'}} palettes={mockPalettes}/>
    )
    mockSetLoadedProject = jest.fn()
    });

  describe('ProjectCard Component', () => {
    it.skip('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });

    // **Write: findProjectPalette, returnElements, createPaletteId, updatedSelectedPalette tests here**
  });

  describe('mapStateToProps', () => {
    it('should return an object with palettes', () => {
      const mockState = {
        projects: [],
        isLoading: false,
        error: '',
        palettes: [],
        extra: '',
        lockedHexcodes: [],
        currentPaletteId: 1,
        randomHexcodes: [],
        shouldDisplayRandom: true
      }
      const expected = {
        palettes: []
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when setLoadedProject is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = setLoadedProject();
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setLoadedProject()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
});