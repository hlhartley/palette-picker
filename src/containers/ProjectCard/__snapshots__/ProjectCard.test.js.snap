// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard ProjectCard Component should properly render the component elements 1`] = `
<div
  className="ProjectCard"
>
  <div
    onClick={[Function]}
  >
    <p
      className="project-name"
    >
      PROJECT1
    </p>
    <div
      className="palette-container"
    >
      <Connect(PaletteCard)
        projectPalettes={
          Array [
            Object {
              "color1": "FFFF82",
              "color2": "DDDCF7",
              "color3": "B5D99C",
              "color4": "FFAF66",
              "color5": "E65F5C",
              "created_at": "2019-03-26T05:17:02.735Z",
              "id": 1,
              "name": "Summer colors",
              "project_id": 1,
              "updated_at": "2019-03-26T05:17:02.735Z",
            },
            Object {
              "color1": "F25469",
              "color2": "BD2D87",
              "color3": "D664BE",
              "color4": "DF99F0",
              "color5": "B191FF",
              "created_at": "2019-03-26T05:17:02.736Z",
              "id": 2,
              "name": "Spring colors",
              "project_id": 1,
              "updated_at": "2019-03-26T05:17:02.736Z",
            },
          ]
        }
      />
    </div>
    <button
      className="delete-project-button"
      onClick={[Function]}
    >
      Delete Project
    </button>
  </div>
</div>
`;
